src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -291,30 +289,36 @@ static ParsedSQLCacheItem parseAndCacheSQL(Sha1HashKey key,
     private int statementPoolingCacheSize = DEFAULT_STATEMENT_POOLING_CACHE_SIZE;
 
     /** Cache of prepared statement handles */
-    private ConcurrentLinkedHashMap<Sha1HashKey, PreparedStatementHandle> preparedStatementHandleCache;
+    private ConcurrentLinkedHashMap<CityHash128Key, PreparedStatementHandle> preparedStatementHandleCache;
     /** Cache of prepared statement parameter metadata */
-    private ConcurrentLinkedHashMap<Sha1HashKey, SQLServerParameterMetaData> parameterMetadataCache;
+    private ConcurrentLinkedHashMap<CityHash128Key, SQLServerParameterMetaData> parameterMetadataCache;
     /**
      * Checks whether statement pooling is enabled or disabled. The default is set to true;
      */
     private boolean disableStatementPooling = true;
 
-    /**
-     * Find statement parameters.
-     * 
-     * @param sql
-     *            SQL text to parse for number of parameters to intialize.
-     */
-    private static int countParams(String sql) {
-        int nParams = 0;
+     /**
+      * Locate statement parameters.
+      * 
+      * @param sql
+      *          SQL text to parse for positions of parameters to intialize.
+      */
+    private static int[] locateParams(String sql) {
+        List<Integer> parameterPositions = new ArrayList<Integer>(0);
```

Comment:
Template type inferred from LHS, not necessary to declare it a second time.

Reply:
List<Integer> parameterPositions = new LinkedList<>(); would be better imo